projects/extension/src/background/ConnectionManager.ts

```diff
@@ -117,9 +117,12 @@ export class ConnectionManager extends (EventEmitter as { new(): StateEmitter })
       port.postMessage({ type: 'info', payload: `App ${port.name} already exists.` });
     } else {
       const newApp = new AppMediator(port, this as ConnectionManagerInterface)
-      this.#apps.push(newApp);
-      newApp.on('stateChanged', () => this.emit('stateChanged'));
-      this.emit('stateChanged');
+      const didAssociate = newApp.associate();
```

Comment:
no need to assign variable and then check against it. you can just:
```
if (newApp?.associate()) {
```

Reply:
True.. this is a remnant of refactoring

Reply:
Yeap. Not a blocker for merging. I requested changed for the lint issues mainly